### ISSUE TITLE
Detect PEP 585 and "with" statement grouping parentheses

### ISSUE THREAD

**Issue by ashegoulding** [netromdk/vermin#84]
The current version of Vermin cannot detect following cases.

* PEP 585: Type Hinting
* "with" statement with grouping parentheses

```python3
import io

my_list = list[str]()

with (
	io.FileIO("/dev/stdin") as a,
	io.FileIO("/dev/stdout", "w") as b):
	while b.write(a.read(4096)): pass
```

`vermin -vv -` output:

```
Detecting python files..
Analyzing using 8 processes..
2.7, 3.1     
  'io' module requires 2.6, 3.0
  `with` requires 2.5, 3.0
  multiple context expressions in a `with` statement requires 2.7, 3.1

Tip: Since '# novm' or '# novermin' weren't used, a speedup can be achieved using: --no-parse-comments
```

Expected output:

```
  Type Hinting Generics requires 3.9
  `with` requires 2.5, 3.0
  multiple context expressions grouped in parentheses in a `with` statement requires 3.10
```

idk what else Vermin is missing.

**Comment by netromdk** [comment]
Thanks for the feedback!

If you use `--eval-annotations` the results are instead:
```
'io' module requires 2.6, 3.0
`with` requires 2.5, 3.0
builtin generic type annotation (list[..]) requires !2, 3.9
multiple context expressions in a `with` statement requires 2.7, 3.1
```
Information about `--eval-annotations` can be found in the [Caveats](https://github.com/netromdk/vermin#caveats) section of the README.

Regarding that
> "with" statement with grouping parentheses

requires Python 3.10+ is, unfortunately, a little more complicated. The code you provided also works in Python 3.9 but not in 3.8. This is due to the difficulty via the abstract syntax tree (AST) to differentiate between
```py
with io.FileIO("/dev/stdin") as a, io.FileIO("/dev/stdout", "w") as b:
  pass
```
and
```py
with (
  io.FileIO("/dev/stdin") as a,
  io.FileIO("/dev/stdout", "w") as b):
  pass
```
The first works in 3.1+ while the second works in 3.9+ but they both look the same in the AST.

I will look into this and check for possible solutions.

**Comment by netromdk** [comment]
Heads up: I've been working on a fix on branch `issue-84`.
The new result becomes:
```
!2, 3.9      /tmp/test.py
  L1 C7: 'io' module requires 2.6, 3.0
  L3: builtin generic type annotation (list[..]) requires !2, 3.9
  L5: `with` requires 2.5, 3.0
  L5: multiple context expressions in a `with` statement requires 2.7, 3.1
  L5: multiple context expressions in a `with` statement with parentheses requires !2, 3.9
```

I also found some other things to complete before merging to `master`.

**Comment by netromdk** [comment]
And I implemented a heuristic to show a tip if unevaluated generic/literal annotations might be in use:
```
Tip: Generic or literal annotations might be in use. If so, try using: --eval-annotations
But check the caveat section: https://github.com/netromdk/vermin#caveats
```
It would be shown when not using `--eval-annotations` , not having tips disabled, and using generic or literal annotations like your input example.

**Comment by netromdk** [comment]
Has been included in release [v1.4.0](https://github.com/netromdk/vermin/releases/tag/v1.4.0) 🎉
